chore(form-system): convert docs and demos to updated storybook (#456)
ing-bank/lion: packages/form-system/stories/01-overview.stories.mdx

import { Story, Meta, html } from '@open-wc/demoing-storybook';
import { render } from '@lion/core';
import '@lion/input/lion-input.js';
import { Validator } from '@lion/validate';
import './helper-wc/h-output.js';

<Meta title="Form Fundaments/Overview" />

# Form System Overview

This page should be used as a starting point when first using the Form System.
It provides an overview of its essential building blocks and provides links to detailed explanations
of most of its core concepts.

## Building Blocks

> Our Form System is built from a set of very fundamental building blocks: `form control`s, `field`s and `fieldset`s.

Reviewer: Nitpick:

Having the final s for the plural form looks funky when it renders.
`form control`s, `field`s and `fieldset`s

What do you think about putting the "s" inside the code block?

```suggestion
Our Form System is built from a set of very fundamental building blocks: `form controls`, `fields` and `fieldsets`.
```

Reviewer 2: The main reason was to highlight the core concepts/defintions in our System. In that sense 'form control' is a concept, 'form controls' not really.
To me this looks better and I think this is the common way to do it. But if you insist on changing it, feel free to do so ;)

### Form Controls

> `Form control`s are the most fundamental building block of our Form System.

Reviewer: What do you think about:

```suggestion
`Form control`s are the fundamental building blocks of our Form System.
```

Something can be fundamental or not. There is no half way fundamental.

Reviewer 2: yep, block should be plural 👍

Reviewer 2: Wrt fundamental. I think it can have various adjectives: 'most fundamental', 'very fundamental', 'quite fundamental' are all very commonly used.
In our particular case we have FormControl, Field and Input.
I would say FormControl is the most fundamental here, since it is shared by __al__l form components, whereas Field is shared by __many__ components.

> They are the fundament of both `field`s, and `fieldset`s and provide a normalized, predictable api

Reviewer: ```suggestion
They are the base of both `fields`, and `fieldsets` and provide a normalized, predictable interface
```

or 

```suggestion
They are at the core of both `fields`, and `fieldsets` and provide a normalized, predictable interface
```

Reviewer 2: All terms above are interchangeable to me; feel free to update ;)

throughout the whole form.
Every form element inherits from `FormControlMixin`.

`FormControlMixin` creates the default html structure

> and accessibility is designed to be used in conjunction with the ValidateMixin and the FormatMixin.

Reviewer: Lines 24 to 27 could be rewritten as:

```suggestion
Form elements inherit from `FormControlMixin`. This mixin is responsible with defining the accessible html structure. It should be used in conjunction with `ValidateMixin ` and `FormatMixin`.
```

Reviewer 2: Language as chosen here was quite on purpose. They are designed to be used in conjunction with, but nothing is obliged. For instance, in the Fieldset we don't do this ourselves for historical reasons. We do however match the exact apis of those mixins in the Fieldset

### Fields

Fields (think of an input, texarea, select) are the actual form controls the end user interacts with.

> They extend `LionField`, which on its turn uses the `FormControlMixin`.

Reviewer: ```suggestion
They extend `LionField`, which in turn uses the `FormControlMixin`.
```

Reviewer 2: 👍

Fields provide a normalized api for both platform components and custom made form controls.

> On top of this, they feature:

Reviewer: ```suggestion
On top of this, they provide / offer:
```

Reviewer 2: maybe make it 'offer' then?

- [formatting/parsing/serializing](?path=/docs/form-fundaments-formatting-and-parsing) of view values
- advanced [validation](?path=/docs/form-fundaments-validation) possibilities
- creation of advanced user interaction scenarios via [interaction states](?path=/docs/form-fundaments-interaction-states--interaction-states)
- provision of labels and help texts in an easy, declarative manner
- accessibility out of the box <!-- TODO: write how our form system manages to deliver accessible components with all challenges that shadow dom provide -->
- advanced styling possibilities: map your own Design System to the internal HTML structure <!-- TODO: link to example of Material Design Datepicker-->

#### Platform fields (wrappers)

Platform field wrapper add all of the functionality described above to form elements natively

> supported by the platform.

Reviewer: ```suggestion
Platform field wrappers add all of the functionality described above to native form elements.
```

Reviewer 2: Feel free to update

- [LionInput](?path=/docs/forms-input), a wrapper for `<input>`
- [LionTextarea](?path=/docs/forms-textarea), a wrapper for `<textarea>`
- [LionSelect](?path=/docs/forms-select), a wrapper for `<select>`
- [LionRadio](?path=/docs/forms-radio-group), a wrapper for `<input type="radio">`
- [LionCheckbox](?path=/docs/forms-checkbox-group), a wrapper for `<input type="checkbox">`
- [LionInputRange](?path=/docs/forms-input-range), a wrapper for `<input type="range">`

### Dedicated fields

Dedicated fields are less generic fields in a sense that they by default expect a certain type of
modelValue. This means that they have validators, parsers and formatters preconfigured.

- [`LionInputDate`](?path=/docs/forms-input-date)
- [`LionInputDatepicker`](?path=/docs/forms-input-datepicker)
- [`LionInputEmail`](?path=/docs/forms-input-email)
- [`LionInputAmount`](?path=/docs/forms-input-amount)
- [`LionInputIban`](?path=/docs/forms-input-iban)

### Custom fields

> Instead of wrapping native elements, this category of fields addresses custom built

Reviewer: ```suggestion
Instead of wrapping native elements, this category of fields contains custom built
```

Reviewer 2: Both are fine to me :)

form elements.

- [`LionSelectRich`](?path=/docs/forms-select-rich), an advanced (rich) version of `<select>`

One could also think of components like:

- listbox
- combobox
- autocomplete
- etc...

For more information about writing a custom field, please see [How to write a custom field](?path=/docs/form-fundaments-creating-a-custom-field--page)

## Fieldsets

Fieldsets are groups of fields. They can be considered fields on their own as well, since they
partly share the normalized api via `FormControlMixin`.

> Fieldsets are the basis for:

Reviewer: ```suggestion
Fieldsets are the base for:
```

Reviewer 2: 👍
